### PR TITLE
Jp vanishing gradients camera processing

### PR DESCRIPTION
**Pull request by blacksaturn1**

### CODE REVIEW

File: final-project/src/tier1/action_primitives.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def straight_line_rotations(self):
         matrix[0] = np.matrix(
             [[0.0, 0.0],
             [0.0, 0.0],
-            [0.0, 1.0]]
```
Comment (arjangupta) [inline final-project/src/tier1/action_primitives.py:36]: Why change this to 1.425?

File: final-project/src/tier1/action_primitives.py [hunk 2]
```diff
@@ -44,6 +44,8 @@ def straight_line_rotations(self):
                 print("matrix" + str(i+1))
                 print(matrix[i])
         # Convert to bezier curve
+        matrix[matrix<-1]=-1
+        matrix[matrix>1]=1
```
Comment (arjangupta) [inline final-project/src/tier1/action_primitives.py:48]: Isn't this just undoing your 1.425 change?

File: final-project/src/tier1/action_primitives.py [hunk 3]
```diff
@@ -64,6 +66,8 @@ def curved_line_rotations(self):
                 print("matrix" + str(i+10))
                 print(matrix[i])
         # Convert to bezier curve
+        matrix[matrix<-1]=-1
+        matrix[matrix>1]=1
```
Comment (arjangupta) [inline final-project/src/tier1/action_primitives.py:70]: Again not sure the point of the 1.425 and these

File: final-project/src/tier1/main.py [hunk 4]
```diff
@@ -47,55 +45,68 @@ def __init__(self, debug=False):
...
+        self.env.reset_idx([0])
 
     def choose_new_goal_position(self):
         """Chooses a new goal position"""
         # Choose a random index from the goal position options
         # self.goal_position = self.goal_position_options[
         #     torch.randint(0, self.goal_position_options.shape[0], (1,)).item()]
         # Just choose a coordinate along the x-axis
-        self.goal_position = torch.tensor([19.0, 0.0, 7.5],
+        self.goal_position = torch.tensor([19.0, 19.0, .2],
```
Comment (arjangupta) [inline final-project/src/tier1/main.py:56]: Probably needs to be reverted to a point along the X axis. Z coord can be whatever

File: final-project/src/tier1/main.py [hunk 5]
```diff
@@ -47,55 +45,68 @@ def __init__(self, debug=False):
...
         points = self.action_primitives.get_sampled_curve(action, num_samples=5)
         # Step through all points
         reset = False
         collision = False
-        for i_sample in range(0, num_samples):
+        for i_sample in range(4, num_samples):
             # Set command actions
             self.command_actions = torch.from_numpy(points[:, i_sample])
-            for _ in range(0, 15):
+            for _ in range(0, 1):
```
Comment (arjangupta) [inline final-project/src/tier1/main.py:81]: If you're going to have these inner and outer loops just execute once, why not just remove them?

And are you having them execute only once because the control step per env step is now 100?

File: final-project/src/tier1/main.py [hunk 6]
```diff
@@ -2,22 +2,20 @@
...
 from aerial_gym.utils import get_args, task_registry
 import torch
-from aeriel_robot_cfg_final_project import AerialRobotCfgFinalProjectTier1
-from aeriel_robot_final_project import AerialRobotFinalProjectTier1
 from action_primitives import QuadActionPrimitives
 from deep_q_learning import DeepQLearningAgent, State
 from reward_system import QuadRewardSystem
+from envs import *
 
 class GymInterface:
```
Comment (arjangupta) [inline final-project/src/tier1/main.py:10]: This class probably should be in its own file too, since you've moved all other classes to separate files. main.py can just combine all that.

Comment (arjangupta) [reply final-project/src/tier1/main.py:10]: This one I guess you don't have to take care of within this PR.

File: final-project/src/tier1/envs/aerial_robot_finalproject_tier1_task.py [hunk 7]
```diff
@@ -402,12 +435,14 @@ def compute_quadcopter_reward(root_positions, root_quats, root_linvels, root_ang
...
     reset = torch.where(progress_buf >= max_episode_length - 1, ones, die)
-    reset = torch.where(torch.norm(root_positions, dim=1) > 20.0, ones, reset) # out of bounds for a norm distance of 20.0
-
+    print("reset1",reset)
+    reset = torch.where(torch.norm(root_positions, dim=1) > 35.0, ones, reset) # out of bounds for a norm distance of 20.0
+    print("reset2",reset)
     # Above a certain self.counter number, if the z coordinate is too close to ground, then reset
-    if counter > 500:
-        ground_threshold = 0.25
+    if counter > -1:
```
Comment (arjangupta) [inline final-project/src/tier1/envs/aerial_robot_finalproject_tier1_task.py:442]: The problem we with this is that sometimes when the drone starts too low the ground, the episode immediately terminates.

Also, sorry but where is ground_threshold initialized? I know it's somewhere but I can't find it for some reason

**Review by arjangupta — CHANGES_REQUESTED** [review]
I can't seem to find the new logic for the controller following the sampled points of the action curves. Where is that now?

File: final-project/src/tier1/main.py (hunk 5 above)
Comment (tberg1234) [reply final-project/src/tier1/main.py:81]: Agreed, I am also a bit confused why we want to run the outer loop once

File: final-project/src/tier1/main.py (hunk 4 above)
Comment (tberg1234) [reply final-project/src/tier1/main.py:56]: To add on, the reasoning behind along the x axis is because of the orientation of the maps. Along the x axis means the goal is at the end of the maze

File: final-project/src/tier1/envs/aerial_robot_finalproject_tier1_cfg.py [hunk 8]
```diff
@@ -66,10 +66,10 @@ class robot_asset:
     class viewer:
         ref_env = 0
         pos = [-5, -5, 4]  # [m]
-        lookat = [0, 0, 0]  # [m]
+        lookat = [1, 1, 1]  # [m]
```
Comment (tberg1234) [inline final-project/src/tier1/envs/aerial_robot_finalproject_tier1_cfg.py:69]: Why did we change this?

File: final-project/src/tier1/envs/aerial_robot_finalproject_tier1_task.py [hunk 9]
```diff
@@ -106,6 +108,10 @@ def __init__(self, cfg: AerialRobotCfg, sim_params, physics_engine, sim_device,
         self.drone_hit_ground_buf = torch.ones(self.num_envs, device=self.device, dtype=torch.long)
 
         self.depth_image = torch.zeros((1, 1024), device=self.device)
+        self.contact_force_tensor = self.gym.acquire_net_contact_force_tensor(self.sim)
```
Comment (tberg1234) [inline final-project/src/tier1/envs/aerial_robot_finalproject_tier1_task.py:111]: I am fine leaving this in here but the reason it wasn't was because tier1 wasn't supposed to have objects

**Review by tberg1234 — CHANGES_REQUESTED** [review]
Once the couple of comments of mine are addressed we can merge. Nothing major

File: final-project/src/tier1/action_primitives.py (hunk 1 above)
Comment (blacksaturn1) [reply final-project/src/tier1/action_primitives.py:36]: The resulting point wasn't reaching 1.0, so I made it greater than 1 or less then -1, so I could clip at 1 or -1.

File: final-project/src/tier1/action_primitives.py (hunk 2 above)
Comment (blacksaturn1) [reply final-project/src/tier1/action_primitives.py:48]: This does the clipping for values >1 or <-1, which is the max distance.

File: final-project/src/tier1/action_primitives.py (hunk 3 above)
Comment (blacksaturn1) [reply final-project/src/tier1/action_primitives.py:70]: Clipping last end point to either 1.0 or -1.0

File: final-project/src/tier1/main.py (hunk 4 above)
Comment (blacksaturn1) [reply final-project/src/tier1/main.py:56]: I had tested this and it learned to do the diagonal, I also understand the setpoint is not being set on diagonal and will revert to x-axis as that is also under testing.

File: final-project/src/tier1/main.py (hunk 5 above)
Comment (blacksaturn1) [reply final-project/src/tier1/main.py:81]: This was a hack to set the moving location directly to the intermediate goal pointbased on what the action would yield as displayed in the figure.  We have an action item to pass in a dynamic list of intermediate points and to make the pre-control code perform the necessary loops.  So, this will go away in next work item.

File: final-project/src/tier1/main.py (hunk 6 above)
Comment (blacksaturn1) [reply final-project/src/tier1/main.py:10]: Yes, left for next PR.

File: final-project/src/tier1/envs/aerial_robot_finalproject_tier1_task.py (hunk 7 above)
Comment (blacksaturn1) [reply final-project/src/tier1/envs/aerial_robot_finalproject_tier1_task.py:442]: I am just setting ground threshold everytime to ground_threshold = 0.15.  The lowest it sets at initiation is or should be .20.  
Issue is that sometimes it goes below or negative and sometimes it clamps down to .2 - need to track down in Aerial Gym or Isaac.

File: final-project/src/tier1/action_primitives.py (hunk 1 above)
Comment (arjangupta) [reply final-project/src/tier1/action_primitives.py:36]: I think we need a comment for that since it is very context dependent.

File: final-project/src/tier1/main.py (hunk 4 above)
Comment (arjangupta) [reply final-project/src/tier1/main.py]: Created https://github.com/WPI-RL/drl_quadcopter/issues/14 for it.

File: final-project/src/tier1/main.py (hunk 5 above)
Comment (arjangupta) [reply final-project/src/tier1/main.py:81]: Please make a task or something so that we can track that it will be gone in the next PR.

**Review by arjangupta — APPROVED** [review]

File: final-project/src/tier1/envs/aerial_robot_finalproject_tier1_cfg.py (hunk 8 above)
Comment (tberg1234) [reply final-project/src/tier1/envs/aerial_robot_finalproject_tier1_cfg.py:69]: Why do we change that?

Comment (tberg1234) [reply final-project/src/tier1/envs/aerial_robot_finalproject_tier1_cfg.py:69]: Why do we change that?

**Review by tberg1234 — APPROVED** [review]
Approved
